File: RecipeShare/Data/Recipe.cs
namespace RecipeShare.Data
{
    public class Recipe
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Instructions { get; set; }
        public int PrepTimeMinutes { get; set; }
        public int CookTimeMinutes { get; set; }
        public int Servings { get; set; }
        public string ImageUrl { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        public List<Ingredient> Ingredients { get; set; }

        public int DifficultyLevelId { get; set; }

        public DifficultyLevel DifficultyLevel { get; set; } = null!;

        public ICollection<DietaryTag> DietaryTags { get; set; } = new List<DietaryTag>();
    }
}

File: RecipeShareAngularApp/src/app/models/recipe.ts
export interface Ingredient {
    id: number;
    name: string;
    amount: string;
    unit: string;
}

export interface Recipe {
    id: number;
    title: string;
    description: string;
    instructions: string;
    prepTimeMinutes: number;
    cookTimeMinutes: number;
    servings: number;
    imageUrl?: string;
    createdAt: Date;
    updatedAt: Date;
    dietaryTags: (string | { id: number; name: string; displayName: string; description: string; recipes: any[] })[];
    difficultyLevel: string;
    difficultyLevelId: number;
    ingredients: Ingredient[];
}

export interface CreateRecipe {
    title: string;
    description: string;
    instructions: string;
    prepTimeMinutes: number;
    cookTimeMinutes: number;
    servings: number;
    imageUrl?: string;
    dietaryTagIds: number[];
    ingredients: CreateIngredient[];
    difficultyLevel: string;
    difficultyLevelId: number;
}

export interface CreateIngredient {
    name: string;
    amount: string;
    unit: string;
}

export interface PaginatedResult<T> {
    items: T[];
    totalCount: number;
    pageNumber: number;
    pageSize: number;
    totalPages: number;
}

export interface RecipeListViewModel {
    id: number;
    title: string;
    description: string;
    prepTimeMinutes: number;
    cookTimeMinutes: number;
    servings: number;
    difficultyLevel: string;
    dietaryTags: string[];
    imageUrl?: string;
}

export interface RecipeDetailViewModel extends RecipeListViewModel {
    ingredients: IngredientViewModel[];
    instructions: string[];
    createdAt: Date;
    updatedAt: Date;
}

export interface IngredientViewModel {
    id: number;
    name: string;
    amount: string;
    unit: string;
}

export interface CreateRecipeViewModel {
    title: string;
    description: string;
    prepTimeMinutes: number;
    cookTimeMinutes: number;
    servings: number;
    difficultyLevel: string;
    dietaryTags: string[];
    ingredients: IngredientViewModel[];
    instructions: string[];
    imageUrl?: string;
}

export interface UpdateRecipeViewModel extends CreateRecipeViewModel {
    id: number;
}
